### PR TITLE
add doc for adler32 transform

### PR DESCRIPTION
**Pull request by stevendborrelli**
Add new transform from upstream PR https://github.com/crossplane/crossplane/pull/4417.

### CODE REVIEW

**Review by jeanduplessis — APPROVED** [review]

**Comment by jeanduplessis** [comment]
We can merge this PR once the XP PR is merged.

**Comment by stevendborrelli** [comment]
@plumbis the upstream PR hasn't been merged yet.

**Comment by plumbis** [comment]
🤦‍♂️. Revert PR #526

**Pull request merged**
